pyfstat/mcmc_based_searches.py

```diff
@@ -3100,181 +3113,6 @@ def _get_p0_per_stage(self, reset_p0=False):
-                )
-            )
-
-            if plot_walkers:
-                try:
-                    walkers_fig, walkers_axes = self._plot_walkers(
-                        nprod=nprod, xoffset=nsteps_total, **walker_plot_args
-                    )
-                    for ax in walkers_axes[: self.ndim]:
-                        ax.axvline(nsteps_total, color="k", ls="--", lw=0.25)
-                except Exception as e:
-                    logging.warning("Failed to plot walkers due to Error {}".format(e))
-
-            nsteps_total += nburn + nprod
-
-        if plot_walkers:
-            nstep_list = np.array(
-                [el[0][0] for el in run_setup] + [run_setup[-1][0][1]]
-            )
-            mids = np.cumsum(nstep_list) - nstep_list / 2
-            mid_labels = ["{:1.0f}".format(i) for i in np.arange(0, len(mids) - 1)]
-            mid_labels += ["Production"]
-            for ax in walkers_axes[: self.ndim]:
-                axy = ax.twiny()
-                axy.tick_params(pad=-10, direction="in", axis="x", which="major")
-                axy.minorticks_off()
-                axy.set_xlim(ax.get_xlim())
-                axy.set_xticks(mids)
-                axy.set_xticklabels(mid_labels)
-
-        samples = self.sampler.chain[0, :, nburn:, :].reshape((-1, self.ndim))
-        lnprobs = self.sampler.logprobability[0, :, nburn:].reshape((-1))
-        lnlikes = self.sampler.loglikelihood[0, :, nburn:].reshape((-1))
-        all_lnlikelihood = self.sampler.loglikelihood
-        self.samples = samples
-        self.lnprobs = lnprobs
-        self.lnlikes = lnlikes
-        self.all_lnlikelihood = all_lnlikelihood
-        if save_pickle:
-            self._pickle_data(samples, lnprobs, lnlikes, all_lnlikelihood)
-        if export_samples:
-            self.export_samples_to_disk()
-        if save_loudest:
-            self.generate_loudest()
-
-        if plot_walkers:
-            try:
-                walkers_fig.tight_layout()
-            except Exception as e:
-                logging.warning(
-                    "Failed to set tight layout for walkers plot due to Error {}".format(
-                        e
-                    )
-                )
-            if (walker_plot_args.get("fig") is not None) and (
-                walker_plot_args.get("axes") is not None
-            ):
-                self.walkers_fig = walkers_fig
-                self.walkers_axes = walkers_axes
-            else:
-                try:
-                    walkers_fig.savefig(
-                        os.path.join(self.outdir, self.label + "_walkers.png")
-                    )
-                    plt.close(walkers_fig)
-                except Exception as e:
-                    logging.warning(
-                        "Failed to save walker plots due to Error {}".format(e)
-                    )
-
 
 class MCMCTransientSearch(MCMCSearch):
     """MCMC search for a transient signal using ComputeFstat
```

Comment:
this has all the parameters duplicated from basic `MCMCSearch`, and the attributes are missing documentation again

Reply:
That's odd... I'm not sure why it happens in this class but not the other ones.

Reply:
Ah, because it doesn't have an overriden `__init__()` and hence inherits it from the main `MCMCSearch` class, full documentation included. That's good then, no duplicate text to maintain!

Reply:
...then the two lines below about "See parent" can be removed, though.